libs/langchain/langchain/text_splitter.py

```diff
@@ -284,20 +266,29 @@ class CharacterTextSplitter(TextSplitter):
     def __init__(
         self, separator: str = "\n\n", is_separator_regex: bool = False, **kwargs: Any
     ) -> None:
-        """Create a new TextSplitter."""
+        """Create a new TextSplitter.
+
+        Args:
+            separator: Separator to split on. If empty string, split on space.
+            is_separator_regex: Indicator whether the separator is a regex.
+        """
         super().__init__(**kwargs)
         self._separator = separator
         self._is_separator_regex = is_separator_regex
 
+        if not self._separator:
+            self._separator = " "
+        # If the separator is a regex, we don't need to escape it.
+        if not self._is_separator_regex:
+            self._separator_regex = re.escape(self._separator)
+        # If we want to keep the separator, we need to capture it.
+        if self._keep_separator:
+            self._separator_regex = f"({self._separator_regex})"
+
     def split_text(self, text: str) -> List[str]:
         """Split incoming text and return chunks."""
-        # First we naively split the large input into a bunch of smaller ones.
-        separator = (
-            self._separator if self._is_separator_regex else re.escape(self._separator)
-        )
-        splits = _split_text_with_regex(text, separator, self._keep_separator)
-        _separator = "" if self._keep_separator else self._separator
-        return self._merge_splits(splits, _separator)
+        splits = _split_text_with_regex(text, self._separator)
+        return self._merge_splits(splits)
```

Comment:
Those are pretty heavy refactoring.
Is it possible to support them with unit tests to make sure everything works well?

Reply:
agreed, some unit tests would be great!

Reply:
Definetely yes!
This is still in development